### ISSUE TITLE
First call to Rand.random(100) after a seeded initRand() is always 0

### ISSUE THREAD

**Issue by awr1** [nim-lang/Nim#18658]
Something extremely unusual:

```nim
import std / random

const Max = 100
for i in 1 .. 10:
  var x = initRand(i)
  echo x.rand(Max)
```

Changing `Max` to, say, `99` or `101` gives us reasonably random output, however, leave it as `100` [and we get all zeros.](https://play.nim-lang.org/#ix=3vmT)

**Comment by Clyybber** [comment]
Can't reproduce on my machine (linux, amd64), I get
```
46
8
99
68
43
77
39
59
75
74
```
on the latest devel https://github.com/nim-lang/Nim/commit/6563a685c1076cb5342f65d192b8f2a983220dba

**Comment by awr1** [comment]
@Clyybber Might have been fixed by 76f74fae?

**Comment by awr1** [comment]
If it was accidentally fixed I think it wouldn't hurt to add some tests for this

**Comment by timotheecour** [comment]
> @Clyybber Might have been fixed by 76f74fa?

no, https://github.com/nim-lang/Nim/issues/18456 fixed different bugs

> If it was accidentally fixed I think it wouldn't hurt to add some tests for this

this bug was fixed in https://github.com/nim-lang/Nim/pull/17468, was intentional, and has tests for this
